SeqGAN/dataloader_helper.py

```diff
@@ -0,0 +1,171 @@
+import json
+import random
+import argparse
+import sys
+
+
+
+# Helper functions for converting between text to DataLoader form
+
+def haiku_to_ls(file):
+    '''
+    Makes lists of haikus from haiku file
+    '''
+    with open(file, 'r') as f:
+        full = []
+        haiku = []
+        for line in f:
+            if line == "\n":
+                num_stopwords = 20 - len(haiku)
+                if num_stopwords > 0:
+                    haiku += ["_FILL_"]*num_stopwords
+                    full.append(haiku)
+                haiku = []
+            else:
+                line = line.strip()
+                line = line.split()
+                line.append("_BREAK_")
+                haiku += line
+    return full
+    
+
+
+def obama_to_ls(file):
+    '''
+    Makes list of Obama speach paragraphs from Obama file.
+    '''
+    with open(file, 'r') as f:
+        full = []
+        paragraph = []
+        for line in f:
+            if line != "\n":
+                line = line.strip()
+                line = line.split()
+                if line != [] and len(line) <= 50:
+                    num_stopwords = 50 - len(line)
+                    paragraph = line + ["_FILL_"]*num_stopwords
+                    full.append(paragraph)
+    return full
+
+
+
+def creat_dicts(lines_ls):
+    '''
+    Creates dictionaries for converting between
+    token and integer form for dataloader.
+    '''
+    all_tokens= []
+    for line in lines_ls:
+        for token in line:
+            all_tokens.append(token)
+
+    all_tokens = list(set(all_tokens))
+    random.shuffle(all_tokens)
+    int_to_word, word_to_int = {}, {}
+    integer_form = 1
+
+    for token in all_tokens:
+        word_to_int[token] = integer_form
+        int_to_word[integer_form] = token
+        integer_form += 1
+
+    return word_to_int, int_to_word, integer_form
```

Comment:
integer_form is not really necessary to return, since you can get the same thing from len(word_to_int) or len(int_to_word).